### PR TITLE
[AutoDiff] Require `[parameters ...]` for `{differentiable,linear}_fu…

### PR DESCRIPTION
**Pull request by dan-zheng**
…nction`.

Make the `[parameters ...]` argument always required in the textual
representation for `{differentiable,linear}_function` instructions, instead of
defaulting to all parameter indices when unspecified..

This makes the textual representation consistent with other `[parameters ...]`
arguments in SIL:
- `differentiability_witness_function`
- `differentiability_witness`

### CODE REVIEW

**Comment by dan-zheng** [comment]
@swift-ci Please test tensorflow

File: lib/ParseSIL/ParseSIL.cpp [hunk 1]
```diff
@@ -2071,6 +2071,31 @@ static bool parseAssignOwnershipQualifier(AssignOwnershipQualifier &Result,
   return false;
 }
 
+// SWIFT_ENABLE_TENSORFLOW
+// Parse an index set, prefaced with the given label. Returns true on error.
+static bool parseIndexSet(Parser &P, StringRef label,
```
Comment (rxwei) [inline lib/ParseSIL/ParseSIL.cpp]: Nothing in this function ensures what's parsed is a set. I'd suggest calling it "index list" instead.
```suggestion
static bool parseIndexList(Parser &P, StringRef label,
```

Comment (dan-zheng) [reply lib/ParseSIL/ParseSIL.cpp]: Good point! Done in https://github.com/apple/swift/pull/29015/commits/8fbccc82b53643a94656c57574b62027570acd5b.

**Review by rxwei — APPROVED** [review]

**Comment by dan-zheng** [comment]
@swift-ci Please test tensorflow

**Comment by dan-zheng** [comment]
@swift-ci Please test tensorflow

**Pull request merged**